library/cwm/src/lib/cwm/widget.rb

```diff
@@ -0,0 +1,450 @@
+require "yast"
+
+require "abstract_method"
+
+module CWM
+  # Represent base for any widget used in CWM. It can be passed as "widget" argument. For more
+  # details about usage see {CWM.ShowAndRun}
+  class AbstractWidget
+    include Yast::UIShortcuts
+    include Yast::I18n
+
+    # @return [String] id used for widget
+    attr_accessor :widget_id
+
+    # defines widget type for CWM usage
+    def self.widget_type=(val)
```

Comment:
I used to have it that way, but found it not so nice. Compare
```
class A
  self.widget_type :button

#versus
class A
  def widget_type
     :button
  end
```
I feel it is more like constant and not method.